src/main/java/ee/tuleva/onboarding/withdrawals/WithdrawalEligibilityService.java

```diff
@@ -0,0 +1,27 @@
+package ee.tuleva.onboarding.withdrawals;
+
+import ee.tuleva.onboarding.auth.principal.Person;
+import ee.tuleva.onboarding.epis.EpisService;
+import ee.tuleva.onboarding.user.personalcode.PersonalCode;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class WithdrawalEligibilityService {
+  private final EpisService episService;
+
+  public WithdrawalEligibilityDto getWithdrawalEligibility(Person person) {
+    var fundPensionCalculation = episService.getFundPensionCalculation(person);
+
+    return new WithdrawalEligibilityDto(
+        hasReachedRetirementAge(person), fundPensionCalculation.durationYears());
+  }
+
+  private boolean hasReachedRetirementAge(Person person) {
+    return PersonalCode.getAge(person.getPersonalCode())
```

Comment:
I would consider adding a `default method` into the `Person` interface, so we could just do:
`person.getAge()` instead of `PersonalCode.getAge(person.getPersonalCode())`

Reply:
Couldn't get it working, `@JsonIgnore` on `Person.getAge` seems to also ignore it for `User` that implements `Person` for `/me`